File: epi_judge_python_solutions/parity.py
from test_framework import generic_test


def parity(x):

    result = 0
    while x:
        result ^= x & 1
        x >>= 1
    return result


if __name__ == '__main__':
    exit(generic_test.generic_test_main("parity.py", 'parity.tsv', parity))

File: epi_judge_java_solutions/epi/Parity.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

public class Parity {

  @EpiTest(testDataFile = "parity.tsv")
  public static short parity(long x) {

    short result = 0;
    while (x != 0) {
      result ^= (x & 1);
      x >>>= 1;
    }
    return result;
  }

  public static void main(String[] args) {
    System.exit(
        GenericTest
            .runFromAnnotations(args, "Parity.java",
                                new Object() {}.getClass().getEnclosingClass())
            .ordinal());
  }
}
